src/app/clusters/thermostat-server/thermostat-server.cpp

```diff
@@ -1439,7 +1478,6 @@ bool emberAfThermostatClusterCommitPresetsSchedulesRequestCallback(
             return SendResponseAndCleanUp(delegate, endpoint, commandObj, commandPath, imcode::InvalidInState);
```

Comment:
I don't think these various SendResponseAndCleanUp calls do the right thing.  Per spec, failure to commit for the reasons being checked for here is supposed to return an AtomicResponse with a list of attribute statuses, etc.